packages/components/Switch/src/useSwitch.ts

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import { useAsPressable, useKeyProps, useOnPressWithFocus, useViewCommandFocus } from '@fluentui-react-native/interactive-hooks';
+import { SwitchProps, SwitchInfo } from './Switch.types';
+import { memoize } from '@fluentui-react-native/framework';
+import { AccessibilityState, AccessibilityActionEvent, LayoutAnimation, AccessibilityActionInfo } from 'react-native';
+import { InteractionEvent } from '@fluentui-react-native/interactive-hooks';
+
+export const useSwitch = (props: SwitchProps): SwitchInfo => {
+  const defaultComponentRef = React.useRef(null);
+  const {
+    onChange,
+    checked,
+    defaultChecked,
+    accessibilityRole,
+    accessibilityLabel,
+    onAccessibilityTap,
+    onAccessibilityAction,
+    label,
+    labelPosition,
+    componentRef = defaultComponentRef,
+    disabled,
+    ...rest
+  } = props;
+
+  const isDisabled = !!disabled;
+  const initialCheckedState = !!(checked ?? defaultChecked);
+  const [checkedState, setCheckedState] = React.useState(initialCheckedState);
+  const focusRef = isDisabled ? null : componentRef;
+
+  const toggleCallback = React.useCallback(
+    (e: InteractionEvent) => {
+      const newCheckedState = checked !== undefined ? checked : !checkedState;
+      onChange && onChange(newCheckedState, e);
+      LayoutAnimation.configureNext(LayoutAnimation.create(200, LayoutAnimation.Types.linear, LayoutAnimation.Properties.scaleX));
```

Comment:
Should this still run if the Switch's checked value is controlled?

Reply:
And do animations need to play if the Switch's checked value changes via the checked prop?

Reply:
I think it would be good if it still ran even though the value is controlled because it can allow developers to still detect if the Switch has been attempted to be toggled. This can be useful if they want a warning to pop up saying that the Switch cannot be used at the moment or something like that.

Reply:
Honestly I'm not 100% confident on whether or not animations need to play in that situation. However, preferably I would like them too. Although at the moment they do not.